### PR TITLE
⛅ Decrease weather update frequency to 2 minutes

### PR DESCRIPTION
**Pull request by webknjaz**
Ref #167

### CODE REVIEW

**Comment by streetturtle** [comment]
Thanks! I think update frequency should be customizable with default value of two minutes. Ideal value would be number of seconds you actually use the widget divided by 1000. For instance, if you work 8 hours on laptop it would be 8*60*60/1000 ~ 28.
I'll merge your PR and will make these changes.

Thank you for your help and feedback :+1:

**Comment by webknjaz** [comment]
> For instance, if you work 8 hours on laptop it would be 8_60_60/1000 ~ 28

That is only true if you actually turn it off. I, for one, have mine turned on almost all the time, sometimes its uptime would be months long. So it's hard to calculate such a number automatically. Also even if you were to collects stats, they'd change all in an unpredictable manner often.

Besides, does one really need the weather updated every 30s? Do you honestly need this info during every moment of your work in front of the computer?
I'd argue that most folks wouldn't care even if it updated once in 15 min...

**Comment by webknjaz** [comment]
Here's an idea: what if the widget implemented some sort of a leaky bucket and counted how many queries are left? And if the user clicks on the widget and if it doesn't exceed the allowed rate limit, it'd do extra calls but not when the user looks at the icon passively.

File: weather-widget/weather.lua [hunk 1]
```diff
@@ -450,7 +450,11 @@ local function worker(args)
         end)))
 
     -- watch('cat /home/pmakhov/.config/awesome/awesome-wm-widgets/weather-widget/weather.json', 5, update_widget, weather_widget)
-    watch(string.format(GET_FORECAST_CMD, owm_one_cal_api), 5, update_widget, weather_widget)
+    watch(
+        string.format(GET_FORECAST_CMD, owm_one_cal_api),
+        120,  -- API limit is 1k req/day; day has 1440 min; every 2 min is good
+        update_widget, weather_widget,
+    )
```
Comment (webknjaz) [inline weather-widget/weather.lua:457]: @streetturtle Oops! it looks like the lua parser does not like this and crashes...

Comment (webknjaz) [reply weather-widget/weather.lua:457]: @streetturtle hotfix: https://github.com/streetturtle/awesome-wm-widgets/pull/171

**Pull request merged**